### PR TITLE
make macaroon lifetimes configurable

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by mhilton** [comment]
Note this is intended to fix #121

File: internal/auth/httpauth/httpauth_test.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (s *authSuite) TestAuthorizeWithAdminCredentials(c *qt.C) {
 				MacaroonVerifier: s.oven,
 				ACLManager:       s.aclManager,
 			})
-			httpAuthorizer := httpauth.New(s.oven, authorizer)
+			httpAuthorizer := httpauth.New(s.oven, authorizer, 0)
```
Comment (cmars) [inline internal/auth/httpauth/httpauth_test.go]: Could this be too tight of a timeout and cause timing-dependent test failures?

File: internal/debug/debug_test.go [hunk 2]
```diff
@@ -91,6 +91,7 @@ func newFixture(c *qt.C) *fixture {
 		c.Skip("mgotest disabled")
 	}
 	c.Assert(err, qt.Equals, nil)
+	db.Session.SetSocketTimeout(time.Minute)
```
Comment (cmars) [inline internal/debug/debug_test.go]: Why is this added? Doesn't seem related to macaroon timeouts. Also seems to be unnecessary as one minute is the default per https://godoc.org/gopkg.in/mgo.v2#Session.SetSocketTimeout.

File: store/mgostore/backend_test.go [hunk 3]
```diff
@@ -22,6 +23,7 @@ func TestNewBackend(t *testing.T) {
 	}
 	c.Assert(err, qt.Equals, nil)
 	defer db.Close()
+	db.Session.SetSocketTimeout(time.Minute)
```
Comment (cmars) [inline store/mgostore/backend_test.go]: Seems unnecessary, see comment above.

File: internal/auth/httpauth/httpauth_test.go (hunk 1 above)
Comment (mhilton) [reply internal/auth/httpauth/httpauth_test.go]: We don't test this in this code path.

File: internal/debug/debug_test.go (hunk 2 above)
Comment (mhilton) [reply internal/debug/debug_test.go]: This is necessary because mgotest sets the socket timeout to 1 second. This was proving too short for my testing using the latest mongo docker image. I will add a comment.
